### ISSUE TITLE
☂️ [GEP-7] Shoot Control Plane Migration "Good Case" Scenario Nice To Have Improvements

### ISSUE THREAD

**Issue by plkokanov** [gardener/gardener#5533]
**How to categorize this issue?**

/area control-plane-migration
/kind enhancement
/roadmap internal
/priority 4

**What would you like to be added**:
This issue tracks additional improvements to the [GEP-7 Control Plane Migration](https://github.com/gardener/gardener/blob/master/docs/proposals/07-shoot-control-plane-migration.md) ("good case") scenario that have lower priority.

Enhancements:
  - [x] https://github.com/gardener/gardener/issues/5411
  - [x] ~#4785 - this is not going to be necessary after #4901 if we forbid changes to the shoot spec during control plane migration and control plane migration is allowed only after successful reconciliations.~

Performance Testing:
  - [x] ~Prepare seeds for performance tests~
  - [x] ~Run performance tests by migrating shoot clusters in bulk~

Performance improvements:
  - [x] ~Check if downtime during control plane migration can be improved, by speeding up the process. E.g. some pods can be deployed on the destination seed before the shoot is completely removed from the source seed.~
